### PR TITLE
DEIMOS wavelength calibration

### PR DESCRIPTION
**Pull request by debora-pe**
This is an all-round work to improve the wavelength calibration for DEIMOS. 

The changes are:

- improved wavelength template for gratings: 600ZD, 830G
- Added new ArI, KrI, NeI, XeI arc lines
- PypeIt can now compute arc line FWHM from the lines themselves. This is controlled by the new parset ``fwhm_fromlines``
- When LVM slit-mask is used, PypeIt reduces only detectors 3 and 7, because the other detectors are always empty. 
- Updated docs
- Report for Keck

Companion PR: https://github.com/pypeit/PypeIt-development-suite/pull/139

### CODE REVIEW

**Comment by debora-pe** [comment]
Dev Suite tests pass:

```
 0 passed/ 0 failed/ 0 skipped STARTED mmt_binospec/Multislit_G270 pypeit (without masters)
 0 passed/ 0 failed/ 0 skipped STARTED magellan_fire/FIRE_Echelle pypeit (without masters)
 0 passed/ 0 failed/ 0 skipped STARTED keck_kcwi/bh2_4200 pypeit (without masters)
 0 passed/ 0 failed/ 0 skipped STARTED gemini_gnirs/32_SB_SXD pypeit (without masters)
 0 passed/ 0 failed/ 0 skipped STARTED lbt_mods/MODS1R_Longslit pypeit (without masters)
 0 passed/ 0 failed/ 0 skipped STARTED mmt_binospec/Longslit_G600 pypeit (without masters)
 0 passed/ 0 failed/ 0 skipped STARTED vlt_xshooter/VIS_manual pypeit (without masters)
 0 passed/ 0 failed/ 0 skipped STARTED keck_deimos/830G_M_8500 pypeit (without masters)
 1 passed/ 0 failed/ 0 skipped PASSED  gemini_gnirs/32_SB_SXD pypeit (without masters)
 1 passed/ 0 failed/ 0 skipped STARTED gemini_gnirs/32_SB_SXD pypeit_sensfunc
 2 passed/ 0 failed/ 0 skipped PASSED  vlt_xshooter/VIS_manual pypeit (without masters)
 2 passed/ 0 failed/ 0 skipped STARTED vlt_xshooter/VIS_manual pypeit_coadd_2dspec
 3 passed/ 0 failed/ 0 skipped PASSED  vlt_xshooter/VIS_manual pypeit_coadd_2dspec
 3 passed/ 0 failed/ 0 skipped STARTED keck_deimos/600ZD_M_6500 pypeit (without masters)
 4 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/830G_M_8500 pypeit (without masters)
 4 passed/ 0 failed/ 0 skipped STARTED lbt_mods/MODS2R_Longslit pypeit (without masters)
 5 passed/ 0 failed/ 0 skipped PASSED  gemini_gnirs/32_SB_SXD pypeit_sensfunc
 5 passed/ 0 failed/ 0 skipped STARTED gemini_gnirs/32_SB_SXD pypeit_flux_setup
 6 passed/ 0 failed/ 0 skipped PASSED  gemini_gnirs/32_SB_SXD pypeit_flux_setup
 6 passed/ 0 failed/ 0 skipped STARTED gemini_gnirs/32_SB_SXD pypeit_flux
 7 passed/ 0 failed/ 0 skipped PASSED  gemini_gnirs/32_SB_SXD pypeit_flux
 7 passed/ 0 failed/ 0 skipped STARTED gemini_gnirs/32_SB_SXD pypeit_coadd_1dspec
 8 passed/ 0 failed/ 0 skipped PASSED  gemini_gnirs/32_SB_SXD pypeit_coadd_1dspec
 8 passed/ 0 failed/ 0 skipped STARTED gemini_gnirs/32_SB_SXD pypeit_coadd_2dspec
 9 passed/ 0 failed/ 0 skipped PASSED  gemini_gnirs/32_SB_SXD pypeit_coadd_2dspec
 9 passed/ 0 failed/ 0 skipped STARTED gemini_gnirs/32_SB_SXD pypeit_tellfit
10 passed/ 0 failed/ 0 skipped PASSED  mmt_binospec/Longslit_G600 pypeit (without masters)
10 passed/ 0 failed/ 0 skipped STARTED gemini_gmos/GS_HAM_R400_860 pypeit (without masters)
11 passed/ 0 failed/ 0 skipped PASSED  lbt_mods/MODS1R_Longslit pypeit (without masters)
11 passed/ 0 failed/ 0 skipped STARTED keck_deimos/1200G_M_7750 pypeit (without masters)
12 passed/ 0 failed/ 0 skipped PASSED  gemini_gnirs/32_SB_SXD pypeit_tellfit
12 passed/ 0 failed/ 0 skipped STARTED vlt_xshooter/NIR pypeit (without masters)
13 passed/ 0 failed/ 0 skipped PASSED  keck_kcwi/bh2_4200 pypeit (without masters)
13 passed/ 0 failed/ 0 skipped STARTED keck_deimos/600ZD_tilted pypeit (without masters)
14 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/600ZD_M_6500 pypeit (without masters)
14 passed/ 0 failed/ 0 skipped STARTED gemini_flamingos2/JH_JH pypeit (without masters)
15 passed/ 0 failed/ 0 skipped PASSED  magellan_fire/FIRE_Echelle pypeit (without masters)
15 passed/ 0 failed/ 0 skipped STARTED keck_lris_blue/multi_300_5000_d680 pypeit (without masters)
16 passed/ 0 failed/ 0 skipped PASSED  lbt_mods/MODS2R_Longslit pypeit (without masters)
16 passed/ 0 failed/ 0 skipped STARTED keck_mosfire/J_multi pypeit (without masters)
17 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/1200G_M_7750 pypeit (without masters)
17 passed/ 0 failed/ 0 skipped STARTED keck_deimos/830G_L_8100 pypeit (without masters)
18 passed/ 0 failed/ 0 skipped PASSED  gemini_flamingos2/JH_JH pypeit (without masters)
18 passed/ 0 failed/ 0 skipped STARTED vlt_xshooter/VIS_1x1 pypeit (without masters)
19 passed/ 0 failed/ 0 skipped PASSED  gemini_gmos/GS_HAM_R400_860 pypeit (without masters)
19 passed/ 0 failed/ 0 skipped STARTED gemini_gmos/GS_HAM_R400_860 pypeit standards (without masters)
20 passed/ 0 failed/ 0 skipped PASSED  keck_lris_blue/multi_300_5000_d680 pypeit (without masters)
20 passed/ 0 failed/ 0 skipped STARTED keck_lris_red/multi_1200_9000_d680_1x2 pypeit (without masters)
21 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/600ZD_tilted pypeit (without masters)
21 passed/ 0 failed/ 0 skipped STARTED mmt_mmirs/J_zJ pypeit (without masters)
22 passed/ 0 failed/ 0 skipped PASSED  vlt_xshooter/NIR pypeit (without masters)
22 passed/ 0 failed/ 0 skipped STARTED vlt_xshooter/VIS_2x1 pypeit (without masters)
23 passed/ 0 failed/ 0 skipped PASSED  gemini_gmos/GS_HAM_R400_860 pypeit standards (without masters)
23 passed/ 0 failed/ 0 skipped STARTED gemini_gmos/GS_HAM_R400_860 pypeit_sensfunc
24 passed/ 0 failed/ 0 skipped PASSED  gemini_gmos/GS_HAM_R400_860 pypeit_sensfunc
24 passed/ 0 failed/ 0 skipped STARTED gemini_gmos/GS_HAM_R400_860 pypeit_flux_setup
25 passed/ 0 failed/ 0 skipped PASSED  gemini_gmos/GS_HAM_R400_860 pypeit_flux_setup
25 passed/ 0 failed/ 0 skipped STARTED gemini_gmos/GS_HAM_R400_860 pypeit_flux
26 passed/ 0 failed/ 0 skipped PASSED  gemini_gmos/GS_HAM_R400_860 pypeit_flux
26 passed/ 0 failed/ 0 skipped STARTED gemini_gmos/GS_HAM_R400_860 pypeit_coadd_1dspec
27 passed/ 0 failed/ 0 skipped PASSED  gemini_gmos/GS_HAM_R400_860 pypeit_coadd_1dspec
27 passed/ 0 failed/ 0 skipped STARTED gemini_flamingos2/HK_HK pypeit (without masters)
28 passed/ 0 failed/ 0 skipped PASSED  keck_mosfire/J_multi pypeit (without masters)
28 passed/ 0 failed/ 0 skipped STARTED keck_deimos/830G_LVM_8400 pypeit (without masters)
29 passed/ 0 failed/ 0 skipped PASSED  keck_lris_red/multi_1200_9000_d680_1x2 pypeit (without masters)
29 passed/ 0 failed/ 0 skipped STARTED keck_lris_red/multi_1200_9000_d680_1x2 pypeit (with masters)
30 passed/ 0 failed/ 0 skipped PASSED  mmt_mmirs/J_zJ pypeit (without masters)
30 passed/ 0 failed/ 0 skipped STARTED keck_lris_red/multi_1200_9000_d680 pypeit (without masters)
31 passed/ 0 failed/ 0 skipped PASSED  vlt_xshooter/VIS_1x1 pypeit (without masters)
31 passed/ 0 failed/ 0 skipped STARTED keck_lris_red/multi_600_5000_d560 pypeit (without masters)
32 passed/ 0 failed/ 0 skipped PASSED  keck_lris_red/multi_1200_9000_d680_1x2 pypeit (with masters)
32 passed/ 0 failed/ 0 skipped STARTED keck_lris_red/multi_400_8500_d560 pypeit (without masters)
33 passed/ 0 failed/ 0 skipped PASSED  gemini_flamingos2/HK_HK pypeit (without masters)
33 passed/ 0 failed/ 0 skipped STARTED keck_mosfire/Y_long pypeit (without masters)
34 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/830G_L_8100 pypeit (without masters)
34 passed/ 0 failed/ 0 skipped STARTED keck_mosfire/K_long pypeit (without masters)
35 passed/ 0 failed/ 0 skipped PASSED  vlt_xshooter/VIS_2x1 pypeit (without masters)
35 passed/ 0 failed/ 0 skipped STARTED keck_deimos/830G_M_8100_26 pypeit (without masters)
36 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/830G_LVM_8400 pypeit (without masters)
36 passed/ 0 failed/ 0 skipped STARTED gemini_gmos/GN_HAM_R400_885 pypeit (without masters)
37 passed/ 0 failed/ 0 skipped PASSED  keck_lris_red/multi_1200_9000_d680 pypeit (without masters)
37 passed/ 0 failed/ 0 skipped STARTED keck_lris_red/multi_1200_9000_d680 pypeit (with masters)
38 passed/ 0 failed/ 0 skipped PASSED  keck_mosfire/Y_long pypeit (without masters)
38 passed/ 0 failed/ 0 skipped STARTED mmt_mmirs/K_K pypeit (without masters)
39 passed/ 0 failed/ 0 skipped PASSED  keck_lris_red/multi_600_5000_d560 pypeit (without masters)
39 passed/ 0 failed/ 0 skipped STARTED keck_lris_red/multi_600_5000_d560 pypeit (with masters)
40 passed/ 0 failed/ 0 skipped PASSED  keck_lris_red/multi_400_8500_d560 pypeit (without masters)
40 passed/ 0 failed/ 0 skipped STARTED keck_lris_red/multi_400_8500_d560 pypeit (with masters)
41 passed/ 0 failed/ 0 skipped PASSED  keck_mosfire/K_long pypeit (without masters)
41 passed/ 0 failed/ 0 skipped STARTED keck_lris_blue_orig/long_600_4000_d500 pypeit (without masters)
42 passed/ 0 failed/ 0 skipped PASSED  keck_lris_red/multi_1200_9000_d680 pypeit (with masters)
42 passed/ 0 failed/ 0 skipped STARTED keck_nires/NIRES pypeit (without masters)
43 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/830G_M_8100_26 pypeit (without masters)
43 passed/ 0 failed/ 0 skipped STARTED lbt_luci/LUCI-I pypeit (without masters)
44 passed/ 0 failed/ 0 skipped PASSED  keck_lris_red/multi_400_8500_d560 pypeit (with masters)
44 passed/ 0 failed/ 0 skipped STARTED p200_tspec/TSPEC pypeit (without masters)
45 passed/ 0 failed/ 0 skipped PASSED  keck_lris_red/multi_600_5000_d560 pypeit (with masters)
45 passed/ 0 failed/ 0 skipped STARTED vlt_xshooter/VIS_2x2 pypeit (without masters)
46 passed/ 0 failed/ 0 skipped PASSED  keck_lris_blue_orig/long_600_4000_d500 pypeit (without masters)
46 passed/ 0 failed/ 0 skipped STARTED not_alfosc/grism19 pypeit (without masters)
47 passed/ 0 failed/ 0 skipped PASSED  mmt_mmirs/K_K pypeit (without masters)
47 passed/ 0 failed/ 0 skipped STARTED keck_lris_blue/multi_600_4000_d560 pypeit (without masters)
48 passed/ 0 failed/ 0 skipped PASSED  gemini_gmos/GN_HAM_R400_885 pypeit (without masters)
48 passed/ 0 failed/ 0 skipped STARTED vlt_fors2/300I pypeit (without masters)
49 passed/ 0 failed/ 0 skipped PASSED  keck_nires/NIRES pypeit (without masters)
49 passed/ 0 failed/ 0 skipped STARTED keck_nires/NIRES pypeit_ql
50 passed/ 0 failed/ 0 skipped PASSED  lbt_luci/LUCI-I pypeit (without masters)
50 passed/ 0 failed/ 0 skipped STARTED mdm_osmos/MDM4K pypeit (without masters)
51 passed/ 0 failed/ 0 skipped PASSED  keck_nires/NIRES pypeit_ql
51 passed/ 0 failed/ 0 skipped STARTED magellan_mage/1x1 pypeit (without masters)
52 passed/ 0 failed/ 0 skipped PASSED  p200_tspec/TSPEC pypeit (without masters)
52 passed/ 0 failed/ 0 skipped STARTED keck_lris_red/long_400_8500_longread pypeit (without masters)
53 passed/ 0 failed/ 0 skipped PASSED  vlt_xshooter/VIS_2x2 pypeit (without masters)
53 passed/ 0 failed/ 0 skipped STARTED mmt_mmirs/HK_zJ pypeit (without masters)
54 passed/ 0 failed/ 0 skipped PASSED  keck_lris_red/long_400_8500_longread pypeit (without masters)
54 passed/ 0 failed/ 0 skipped STARTED keck_lris_red/long_400_8500_longread pypeit (with masters)
55 passed/ 0 failed/ 0 skipped PASSED  keck_lris_red/long_400_8500_longread pypeit (with masters)
55 passed/ 0 failed/ 0 skipped STARTED keck_lris_red/long_600_7500_d560 pypeit (without masters)
56 passed/ 0 failed/ 0 skipped PASSED  mmt_mmirs/HK_zJ pypeit (without masters)
56 passed/ 0 failed/ 0 skipped STARTED gemini_gnirs/10_LB_SXD pypeit (without masters)
57 passed/ 0 failed/ 0 skipped PASSED  magellan_mage/1x1 pypeit (without masters)
57 passed/ 0 failed/ 0 skipped STARTED lbt_luci/LUCI-II pypeit (without masters)
58 passed/ 0 failed/ 0 skipped PASSED  mdm_osmos/MDM4K pypeit (without masters)
58 passed/ 0 failed/ 0 skipped STARTED keck_lris_red_orig/long_300_5000 pypeit (without masters)
59 passed/ 0 failed/ 0 skipped PASSED  vlt_fors2/300I pypeit (without masters)
59 passed/ 0 failed/ 0 skipped STARTED shane_kast_red/600_7500_d57 pypeit (without masters)
60 passed/ 0 failed/ 0 skipped PASSED  keck_lris_blue/multi_600_4000_d560 pypeit (without masters)
60 passed/ 0 failed/ 0 skipped STARTED keck_lris_blue/multi_600_4000_d560 pypeit_coadd_2dspec
61 passed/ 0 failed/ 0 skipped PASSED  not_alfosc/grism19 pypeit (without masters)
61 passed/ 0 failed/ 0 skipped STARTED p200_dbsp_red/316_7500_d55 pypeit (without masters)
62 passed/ 0 failed/ 0 skipped PASSED  keck_lris_blue/multi_600_4000_d560 pypeit_coadd_2dspec
62 passed/ 0 failed/ 0 skipped STARTED gemini_gmos/GS_HAM_R400_700 pypeit (without masters)
63 passed/ 0 failed/ 0 skipped PASSED  keck_lris_red/long_600_7500_d560 pypeit (without masters)
63 passed/ 0 failed/ 0 skipped STARTED keck_lris_red/long_600_7500_d560 pypeit (with masters)
64 passed/ 0 failed/ 0 skipped PASSED  gemini_gnirs/10_LB_SXD pypeit (without masters)
64 passed/ 0 failed/ 0 skipped STARTED keck_lris_blue/long_600_4000_d560 pypeit (without masters)
65 passed/ 0 failed/ 0 skipped PASSED  keck_lris_red/long_600_7500_d560 pypeit (with masters)
65 passed/ 0 failed/ 0 skipped STARTED keck_lris_blue/long_400_3400_d560 pypeit (without masters)
66 passed/ 0 failed/ 0 skipped PASSED  lbt_luci/LUCI-II pypeit (without masters)
66 passed/ 0 failed/ 0 skipped STARTED shane_kast_red/600_5000_d46 pypeit (without masters)
67 passed/ 0 failed/ 0 skipped PASSED  keck_lris_red_orig/long_300_5000 pypeit (without masters)
67 passed/ 0 failed/ 0 skipped STARTED shane_kast_blue/600_4310_d55 pypeit_setup
68 passed/ 0 failed/ 0 skipped PASSED  shane_kast_blue/600_4310_d55 pypeit_setup
68 passed/ 0 failed/ 0 skipped STARTED shane_kast_blue/600_4310_d55 pypeit (without masters)
69 passed/ 0 failed/ 0 skipped PASSED  p200_dbsp_red/316_7500_d55 pypeit (without masters)
69 passed/ 0 failed/ 0 skipped STARTED shane_kast_red/1200_5000_d57 pypeit (without masters)
70 passed/ 0 failed/ 0 skipped PASSED  shane_kast_red/600_7500_d57 pypeit (without masters)
70 passed/ 0 failed/ 0 skipped STARTED keck_nirspec/LOW_NIRSPEC-1 pypeit (without masters)
71 passed/ 0 failed/ 0 skipped PASSED  gemini_gmos/GS_HAM_R400_700 pypeit (without masters)
71 passed/ 0 failed/ 0 skipped STARTED shane_kast_blue/452_3306_d57 pypeit (without masters)
72 passed/ 0 failed/ 0 skipped PASSED  keck_lris_blue/long_600_4000_d560 pypeit (without masters)
72 passed/ 0 failed/ 0 skipped STARTED p200_dbsp_blue/600_4000_d55 pypeit (without masters)
73 passed/ 0 failed/ 0 skipped PASSED  shane_kast_blue/600_4310_d55 pypeit (without masters)
73 passed/ 0 failed/ 0 skipped STARTED shane_kast_blue/600_4310_d55 pypeit_sensfunc
74 passed/ 0 failed/ 0 skipped PASSED  shane_kast_blue/600_4310_d55 pypeit_sensfunc
74 passed/ 0 failed/ 0 skipped STARTED shane_kast_blue/600_4310_d55 pypeit_flux_setup
75 passed/ 0 failed/ 0 skipped PASSED  shane_kast_blue/600_4310_d55 pypeit_flux_setup
75 passed/ 0 failed/ 0 skipped STARTED shane_kast_blue/600_4310_d55 pypeit_flux
76 passed/ 0 failed/ 0 skipped PASSED  keck_lris_blue/long_400_3400_d560 pypeit (without masters)
76 passed/ 0 failed/ 0 skipped STARTED shane_kast_blue/830_3460_d46 pypeit (without masters)
77 passed/ 0 failed/ 0 skipped PASSED  shane_kast_blue/600_4310_d55 pypeit_flux
77 passed/ 0 failed/ 0 skipped STARTED shane_kast_blue/600_4310_d55 pypeit_coadd_1dspec
78 passed/ 0 failed/ 0 skipped PASSED  shane_kast_blue/600_4310_d55 pypeit_coadd_1dspec
78 passed/ 0 failed/ 0 skipped STARTED shane_kast_blue/600_4310_d55 pypeit_ql
79 passed/ 0 failed/ 0 skipped PASSED  shane_kast_red/600_5000_d46 pypeit (without masters)
79 passed/ 0 failed/ 0 skipped STARTED magellan_fire/FIRE_Long pypeit (without masters)
80 passed/ 0 failed/ 0 skipped PASSED  shane_kast_blue/452_3306_d57 pypeit (without masters)
80 passed/ 0 failed/ 0 skipped STARTED shane_kast_red/600_7500_d55_ret pypeit (without masters)
81 passed/ 0 failed/ 0 skipped PASSED  keck_nirspec/LOW_NIRSPEC-1 pypeit (without masters)
81 passed/ 0 failed/ 0 skipped STARTED not_alfosc/grism4 pypeit (without masters)
82 passed/ 0 failed/ 0 skipped PASSED  shane_kast_red/1200_5000_d57 pypeit (without masters)
82 passed/ 0 failed/ 0 skipped STARTED gemini_gmos/GN_HAM_NS_B600_620 pypeit (without masters)
83 passed/ 0 failed/ 0 skipped PASSED  shane_kast_blue/600_4310_d55 pypeit_ql
83 passed/ 0 failed/ 0 skipped STARTED mmt_bluechannel/300l pypeit (without masters)
84 passed/ 0 failed/ 0 skipped PASSED  shane_kast_blue/830_3460_d46 pypeit (without masters)
84 passed/ 0 failed/ 0 skipped STARTED keck_deimos/900ZD_M_6000 pypeit (without masters)
85 passed/ 0 failed/ 0 skipped PASSED  magellan_fire/FIRE_Long pypeit (without masters)
85 passed/ 0 failed/ 0 skipped STARTED keck_deimos/1200B_M_5200 pypeit (without masters)
86 passed/ 0 failed/ 0 skipped PASSED  p200_dbsp_blue/600_4000_d55 pypeit (without masters)
86 passed/ 0 failed/ 0 skipped STARTED shane_kast_red/300_7500_Ne pypeit (without masters)
87 passed/ 0 failed/ 0 skipped PASSED  shane_kast_red/600_7500_d55_ret pypeit (without masters)
87 passed/ 0 failed/ 0 skipped STARTED keck_deimos/1200G_M_5500 pypeit (without masters)
88 passed/ 0 failed/ 0 skipped PASSED  not_alfosc/grism4 pypeit (without masters)
88 passed/ 0 failed/ 0 skipped STARTED keck_deimos/1200B_LVM_5200 pypeit (without masters)
89 passed/ 0 failed/ 0 skipped PASSED  mmt_bluechannel/300l pypeit (without masters)
90 passed/ 0 failed/ 0 skipped PASSED  shane_kast_red/300_7500_Ne pypeit (without masters)
91 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/1200B_LVM_5200 pypeit (without masters)
92 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/1200G_M_5500 pypeit (without masters)
93 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/1200B_M_5200 pypeit (without masters)
94 passed/ 0 failed/ 0 skipped PASSED  gemini_gmos/GN_HAM_NS_B600_620 pypeit (without masters)
95 passed/ 0 failed/ 0 skipped PASSED  mmt_binospec/Multislit_G270 pypeit (without masters)
96 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/900ZD_M_6000 pypeit (without masters)

--- PYPEIT DEVELOPMENT SUITE PASSED 96/96 TESTS (Masters ignored) ---
Testing Started at 2021-02-22T21:27:07.336210
Testing Completed at 2021-02-23T03:31:44.969186
Total Time: 6:04:37.632976

```

And unit tests pass too.

```
=========================================== test session starts ============================================
platform darwin -- Python 3.8.5, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /Users/dpelliccia/PypeIt/PypeIt, configfile: setup.cfg
collected 256 items                                                                                        

test_alignments.py .                                                                                 [  0%]
test_arc.py .                                                                                        [  0%]
test_arcimage.py ..                                                                                  [  1%]
test_biasframe.py ..                                                                                 [  2%]
test_bitmask.py ....                                                                                 [  3%]
test_bpmimage.py ..                                                                                  [  4%]
test_bspline.py ..........                                                                           [  8%]
test_buildimage.py .                                                                                 [  8%]
test_calibrations.py .......                                                                         [ 11%]
test_chk_calibs.py ..                                                                                [ 12%]
test_coadd.py ..                                                                                     [ 13%]
test_cooked.py .                                                                                     [ 13%]
test_coremeta.py .                                                                                   [ 14%]
test_datacontainer.py ........                                                                       [ 17%]
test_detector.py ...                                                                                 [ 18%]
test_display.py .                                                                                    [ 18%]
test_edgetrace.py .                                                                                  [ 19%]
test_fitting.py ....                                                                                 [ 20%]
test_flatfield.py .                                                                                  [ 21%]
test_flexure.py .                                                                                    [ 21%]
test_flux.py ....                                                                                    [ 23%]
test_fluxspec.py ..                                                                                  [ 23%]
test_frametype.py .                                                                                  [ 24%]
test_load.py .                                                                                       [ 24%]
test_load_images.py ...........                                                                      [ 28%]
test_maskdesign_matching.py ...                                                                      [ 30%]
test_masterframe.py .                                                                                [ 30%]
test_match.py ....                                                                                   [ 32%]
test_metadata.py ......                                                                              [ 34%]
test_moment.py ......                                                                                [ 36%]
test_msgs.py ..                                                                                      [ 37%]
test_opticalmodel.py ....                                                                            [ 39%]
test_parse.py ...                                                                                    [ 40%]
test_pca.py ........                                                                                 [ 43%]
test_processimage.py ..                                                                              [ 44%]
test_procimg.py .                                                                                    [ 44%]
test_pydl.py .                                                                                       [ 44%]
test_pypeit.py ..                                                                                    [ 45%]
test_pypeitimage.py .                                                                                [ 46%]
test_pypeitpar.py ................................                                                   [ 58%]
test_pypeitsetup.py .......                                                                          [ 61%]
test_qa.py .                                                                                         [ 61%]
test_runpypeit.py ..                                                                                 [ 62%]
test_scienceimage.py ...                                                                             [ 63%]
test_scripts.py .............                                                                        [ 68%]
test_setups.py .....................                                                                 [ 76%]
test_skysub.py ..                                                                                    [ 77%]
test_slitmask.py ..                                                                                  [ 78%]
test_slittrace.py ....                                                                               [ 80%]
test_spec2dobj.py ......                                                                             [ 82%]
test_specobj.py ......                                                                               [ 84%]
test_specobjs.py .....                                                                               [ 86%]
test_spectrographs.py ....................                                                           [ 94%]
test_syncspec.py .                                                                                   [ 94%]
test_traceimage.py .                                                                                 [ 95%]
test_utils.py ...                                                                                    [ 96%]
test_vcorr.py ..                                                                                     [ 97%]
test_wavemodel.py .                                                                                  [ 97%]
test_wavetilts.py ....                                                                               [ 99%]
test_wvcalib.py ..                                                                                   [100%]

===================================== 256 passed in 923.07s (0:15:23) ======================================

```

File: doc/dev/deimos_wavecalib.rst [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+======== ==========================================================================================================
+
+Including six datasets using LVM slitmask, which is a special slitmask composed of 5 slits with different widths
+(0.7", 0.8", 1.", 1.2", 1.5"). These datasets were taken with the following gratings (and central wavelengths):
+830G (8400 |AA|), 600ZD (7100 |AA|), 830G (7900 |AA|), 1200G (7800 |AA|), 1200B (5450 |AA|), 1200B (5200 |AA|).
+
+Results
+^^^^^^^
+
+Out of the **4072 slits** that have been tested, for only **37 (0.9%)** the wavelength calibration failed or
```
Comment (profxj) [inline doc/dev/deimos_wavecalib.rst:137]: Maybe point out that these can be fixed using the `identify` GUI.  Or I hope they can. :)

File: pypeit/core/wavecal/spectrographs/templ_keck_deimos.py [hunk 2]
```diff
@@ -6,31 +6,53 @@
...
-    slits = [0, 1]
-    lcut = [7192.]
-    xidl_file = os.path.join(templates.template_path, 'Keck_DEIMOS', '600ZD', 'deimos_600.sav')
-    outroot = 'keck_deimos_600.fits'
-    templates.build_template(xidl_file, slits, lcut, binspec, outroot, lowredux=True)
+    outroot = 'keck_deimos_600ZD.fits'
+    # PypeIt fits
+    wpath = os.path.join(templates.template_path, 'Keck_DEIMOS', '600ZD')
+
+    basefiles = ['MasterWaveCalib_A_1_02_useS0896.fits', 'MasterWaveCalib_A_1_02_useS0477.fits',
```
Comment (profxj) [inline pypeit/core/wavecal/spectrographs/templ_keck_deimos.py:15]: Have you added these to the DevSuite?  If not, please do.

**Review by profxj — APPROVED** [review]
Great stuff!

**Review by bpholden — COMMENTED** [review]
Do we want to leave old code in the routine that makes the template commented out or should we just delete it? I prefer deletion but if the keeping the history is important, we can keep it. I would suggest adding a comment as to why the old code remains.

**Review by bpholden — APPROVED** [review]

File: pypeit/core/wavecal/spectrographs/templ_keck_deimos.py (hunk 2 above)
Comment (debora-pe) [reply pypeit/core/wavecal/spectrographs/templ_keck_deimos.py:15]: Yes, I did. I actually have a PR for that. https://github.com/pypeit/PypeIt-development-suite/pull/139

File: doc/dev/deimos_wavecalib.rst (hunk 1 above)
Comment (debora-pe) [reply doc/dev/deimos_wavecalib.rst:137]: Done.

**Comment by debora-pe** [comment]
merging.

**Pull request merged**